Incorrect identically-labeled DataFrame objects Exception comes with df.compare method #50083 (#50287)
pandas-dev/pandas: doc/source/whatsnew/v2.0.0.rst

- Added :meth:`Index.infer_objects` analogous to :meth:`Series.infer_objects` (:issue:`50034`)
- Added ``copy`` parameter to :meth:`Series.infer_objects` and :meth:`DataFrame.infer_objects`, passing ``False`` will avoid making copies for series or columns that are already non-object or where no better dtype can be inferred (:issue:`50096`)
- :meth:`DataFrame.plot.hist` now recognizes ``xlabel`` and ``ylabel`` arguments (:issue:`49793`)

> - :func:`DataFrame.compare` exception message thrown improved to include that both index and columns that are identically labeled can be compared. (:issue:`50083`)

Reviewer: ```suggestion
- Improved error message in :func:`DataFrame.compare` to clarify that "identically labelled" refers to both index and columns (:issue:`50083`)
```

Author: I updated the whatsnew note to match your comment above

- Added :meth:`Index.infer_objects` analogous to :meth:`Series.infer_objects` (:issue:`50034`)
- Added ``copy`` parameter to :meth:`Series.infer_objects` and :meth:`DataFrame.infer_objects`, passing ``False`` will avoid making copies for series or columns that are already non-object or where no better dtype can be inferred (:issue:`50096`)
- :meth:`DataFrame.plot.hist` now recognizes ``xlabel`` and ``ylabel`` arguments (:issue:`49793`)

> - Improved error message in :func:`DataFrame.compare` to clarify that "identically labelled" refers to both index and columns (:issue:`50083`)

Reviewer: sorry, just noticed that this affects much more than just `compare`, e.g:
```python
In [8]: df == dfother
---------------------------------------------------------------------------
ValueError: Can only compare identically-labeled DataFrame objects
```
would also have thrown this error

let's rephrase as
```python
Improved error message when trying to align :class:`DataFrame` objects (for example, in :func:`DataFrame.compare`) to clarify ...
```
